plugin/caffe/caffe_loss-inl.h

```diff
@@ -141,7 +141,7 @@ class CaffeLoss : public Operator {
     top_[0]->set_cpu_diff(&grad_scale_);
 
     // Set BP flag
-    for (index_t i = 0; i < param_.num_data; ++i)
+    for (int i = 0; i < param_.num_data; ++i)
       flags_[i] = req[i] != kNullOp;
 
     caffeOp_->Backward(top_, flags_, bot_);
```

Comment:
@HrWangChengdu Please check if you need to call get_cpu/gpu_diff for all the output blobs. Sometimes caffe gpu layers fallback to cpu and the output is only copied to gpu when the next layer calls get_gpu_data/diff.
Same for forward.